src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -192,14 +192,20 @@ internal sealed class SynthesizedParameterSymbol : SynthesizedParameterSymbolBas
             int ordinal,
             RefKind refKind,
             string name = "",
-            ImmutableArray<CustomModifier> refCustomModifiers = default(ImmutableArray<CustomModifier>))
+            ImmutableArray<CustomModifier> refCustomModifiers = default,
+            ImmutableArray<CSharpAttributeData> attributes = default)
         {
-            if (refCustomModifiers.IsDefaultOrEmpty)
+            if (refCustomModifiers.IsDefaultOrEmpty && attributes.IsDefaultOrEmpty)
             {
                 return new SynthesizedParameterSymbol(container, type, ordinal, refKind, name);
             }
 
-            return new SynthesizedParameterSymbolWithCustomModifiers(container, type, ordinal, refKind, name, refCustomModifiers);
+            if (attributes.IsDefault)
+            {
+                attributes = ImmutableArray<CSharpAttributeData>.Empty;
```

Comment:
The base GetAttributes method returns an empty array, and I found some places that assumed that the array would not be 'default', so it seemed like it made more sense to just ensure that the return value of GetAttributes is not 'default'.